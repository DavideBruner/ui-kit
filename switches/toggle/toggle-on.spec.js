import React from 'react';
import { shallow } from 'enzyme';
import ToggleOn from './toggle-on';

const createTestProps = custom => ({
  isDisabled: false,
  isChecked: false,
  ...custom,
});

describe('<ToggleOn>', () => {
  describe('rendering', () => {
    let props;
    let wrapper;

    it('outputs correct tree', () => {
      props = createTestProps();
      wrapper = shallow(<ToggleOn {...props} />);

      expect(wrapper).toMatchSnapshot();
    });

    describe('when checked', () => {
      beforeEach(() => {
        props = createTestProps({ isChecked: true });
        wrapper = shallow(<ToggleOn {...props} />);
      });

      it('renders a <span>', () => {
        expect(wrapper).toRender('span');
      });
    });

    describe('when not checked', () => {
      beforeEach(() => {
        props = createTestProps({ isChecked: false });
        wrapper = shallow(<ToggleOn {...props} />);
      });

      it('should not render any children', () => {
        expect(wrapper.children()).toHaveLength(0);
      });
    });

    describe('when checked and disabled', () => {
      beforeEach(() => {
        props = createTestProps({ isDisabled: true, isChecked: true });
        wrapper = shallow(<ToggleOn {...props} />);
      });

      it('should render a <span> with disabled class', () => {
        expect(wrapper.find('span')).toHaveClassName('textWrapperDisabled');
      });
    });
  });
});
