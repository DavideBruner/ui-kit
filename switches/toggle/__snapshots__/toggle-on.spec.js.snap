// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleOn> rendering outputs correct tree 1`] = `undefined`;
